source/common/http/http2/codec_impl.h

```diff
@@ -265,11 +275,15 @@ class ClientConnectionImpl : public ClientConnection, public ConnectionImpl {
   // Propogate network connection watermark events to each stream on the connection.
   // The router will propogate it downstream.
   void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
```

Comment:
Can we push all this logic into base Connection class? Do we need to override in client? Seems like booleans and asserts would still be useful in the base class and then we can collapse the duplicate code?